recipes/boost/all/test_package/CMakeLists.txt

```diff
@@ -1,89 +1,80 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
This is a huge raise. Is it really needed?
We have an ongoing discussion about the minimum version right now #2184, but I would be surprised if we are ending up with such a high version number


What CMake features are really needed for this test_package?

Reply:
Exactly! 3.1 is enough here.

Reply:
Maybe the cause was this? https://github.com/conan-io/conan-center-index/pull/2096#issuecomment-661325542

CMake 3.16 does not support boost 1.73:
```
CMake Warning at C:/Program Files/CMake/share/cmake-3.16/Modules/FindBoost.cmake:1161 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.16/Modules/FindBoost.cmake:1283 (_Boost_COMPONENT_DEPENDENCIES)
  C:/Program Files/CMake/share/cmake-3.16/Modules/FindBoost.cmake:1921 (_Boost_MISSING_DEPENDENCIES)
  source_subfolder/CMakeLists.txt:12 (find_package)


CMake Error at C:/Program Files/CMake/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find Boost (missing: filesystem) (found version "1.73.0")
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.16/Modules/FindBoost.cmake:2179 (find_package_handle_standard_args)
  source_subfolder/CMakeLists.txt:12 (find_package)
```

Reply:
But `boost/all/test_package/conanfile.py` is using `cmake_find_package`, so it is not needed.

Reply:
CMake 3.1 is ok

Reply:
@klimkin
I'm now boostdep (see https://github.com/conan-io/conan-center-index/pull/2096) using boost/1.70.0  with cmake 3.8.2.
When passing `Boost_ADDITIONAL_VERSIONS=1.70.0` to cmake, the following warning is displayed: 
```
CMake Warning at /tmp/cmake-3.8.2-Linux-x86_64/share/cmake-3.8/Modules/FindBoost.cmake:765 (message):
  Imported targets not available for Boost version 107000
```

Reply:
CMake have hard-coded Boost versions for component and dependency detection. For example, CMake 3.15.2 lists Boost up to 1.70.0 and will emit a warning if greater. The test_package passes for me despite the warning, since the used components are common across versions. The warning goes away if using matching CMake and Boost versions.